Makefile

```diff
@@ -84,6 +84,12 @@ SONIC_BUILD_INSTRUCTION :=  make \
 	@docker inspect --type image $(SLAVE_IMAGE):$(SLAVE_TAG) &> /dev/null || \
 	    { echo Image $(SLAVE_IMAGE):$(SLAVE_TAG) not found. Building... ; \
 	    $(DOCKER_BUILD) ; }
+ifneq ($(USERNAME),)
```

Comment:
>ifneq [](start = 0, length = 5)

Could you check and warn inside slave.mk?

Reply:
Yes, sure. Replace echo with $(warning ...) and improve warning message text. Looks much better now.

However if someone adds USERNAME and PASSWORD to rules/config they will get false positive warning. But I think this isn't big trouble.